lib/interface.py

```diff
@@ -272,14 +272,12 @@ def __init__(self, server, socket, *, max_message_bytes=0):
         self.host, self.port, _ = server.rsplit(':', 2)
         self.socket = socket
 
-        self.pipe = util.SocketPipe(socket, max_message_bytes=max_message_bytes)
-        self.pipe.set_timeout(0.0)  # Don't wait for data
+        self.pipe = util.JSONSocketPipe(socket, max_message_bytes=max_message_bytes)
         # Dump network messages.  Set at runtime from the console.
         self.debug = False
         self.unsent_requests = []
         self.unanswered_requests = {}
         self.last_send = time.time()
```

Comment:
Small nit here -- since we are here anywa: would you mind defining `self.request_time` too at construction?  

Personal nit: I hate it when Python programmers don't define attributes always.  

Rationale: To program defensively and keep this code maintainable.  It doesn't matter *now* the way this class was written -- but someday someone may brain-fart and assume `self.request_time` is always defined when it may not be, if no requests were ever sent.

Reply:
oh good catch, I noticed that too but forgot about it.